### ISSUE TITLE
ERROR: No miner backend

### ISSUE THREAD

**Issue by steamingtracks** [fireice-uk/xmr-stak#2111]
Please provide as much as possible information to reproduce the issue.

# Basic information
  - Type of the CPU.
Intel Celeron
  - Type of the GPU (if you try to miner with the GPU).
N/A
# Compile issues
  - Which OS do you use?
GalliumOS(Ubuntu 14.04)
  ```
  add **all** commands you used and the **full** compile output here
 `
chrx@chrx:~$ cd xmr-stat
bash: cd: xmr-stat: No such file or directory
chrx@chrx:~$ cd chrx
bash: cd: chrx: No such file or directory
chrx@chrx:~$ cd home
bash: cd: home: No such file or directory
chrx@chrx:~$ cd xmr-stak
chrx@chrx:~/xmr-stak$ cd bin
chrx@chrx:~/xmr-stak/bin$ ./xmr-stak
[2018-12-02 11:51:58] : Your CPU doesn't support hardware AES. Don't expect high hashrates.
[2018-12-02 11:51:58] : MEMORY ALLOC FAILED: mmap failed, check attribute 'use_slow_memory' in 'config.txt'
[2018-12-02 11:51:58] : MEMORY ALLOC FAILED: mmap failed, check attribute 'use_slow_memory' in 'config.txt'
[2018-12-02 11:51:58] : MEMORY ALLOC FAILED: mmap failed, check attribute 'use_slow_memory' in 'config.txt'
[2018-12-02 11:51:58] : MEMORY ALLOC FAILED: mmap failed, check attribute 'use_slow_memory' in 'config.txt'
[2018-12-02 11:51:58] : MEMORY ALLOC FAILED: mmap failed, check attribute 'use_slow_memory' in 'config.txt'
-------------------------------------------------------------------
xmr-stak 2.6.0 8713716

Brought to you by fireice_uk and psychocrypt under GPLv3.
Based on CPU mining code by wolf9466 (heavily optimized by fireice_uk).

Configurable dev donation level is set to 2.0%

-------------------------------------------------------------------
You can use following keys to display reports:
'h' - hashrate
'r' - results
'c' - connection
-------------------------------------------------------------------
Upcoming xmr-stak-gui is sponsored by:
   #####   ______               ____
 ##     ## | ___ \             /  _ \
#    _    #| |_/ /_   _   ___  | / \/ _   _  _ _  _ _  ___  _ __    ___  _   _
#   |_|   #|    /| | | | / _ \ | |   | | | || '_|| '_|/ _ \| '_ \  / __|| | | |
#         #| |\ \| |_| || (_) || \_/\| |_| || |  | | |  __/| | | || (__ | |_| |
 ##     ## \_| \_|\__, | \___/ \____/ \__,_||_|  |_|  \___||_| |_| \___| \__, |
   #####           __/ |                                                  __/ |
                  |___/   https://ryo-currency.com                       |___/

This currency is a way for us to implement the ideas that we were unable to in
Monero. See https://github.com/fireice-uk/cryptonote-speedup-demo for details.
-------------------------------------------------------------------
[2018-12-02 11:51:59] : Mining coin: ryo
[2018-12-02 11:51:59] : WARNING: backend CPU disabled.
[2018-12-02 11:51:59] : ERROR: No miner backend enabled.
chrx@chrx:~/xmr-stak/bin$ 
`cmake -DCUDA_ENABLE=OFF -DOpenCL_ENABLE=OFF -DCPU_ENABLE=ON`
^^^
Those were the parameters I used for cmake.
It complains about there being no miner backend, but I'm mining off of my weak cpu. What can I do for this issue?

**Comment by Spudz76** [comment]
Weak CPU will not autoconfig.  You *must* hand-configure...
It should have formed a `cpu.txt` just with no worker-threads defined.  Copy example line remove leading asterisk.

**Comment by psychocrypt** [comment]
please remove cpu.txt and restart the miner again and post the output from
that start

**Comment by Spudz76** [comment]
cpu.txt will have a thread conf section, which is an empty array.

This is normal, it does it on my Atom (512KB cache, no L3...) but when hand forced to two threads it runs some hashes (not fast at all, use autodiff proxy or a pool with autodiff down to 64 or something hilarious).

**Comment by thewalrusking** [comment]
i have a celereon i3,4 amd rx vega 64 and  using windows . and i am also getting this message. i was mining fine a few hours ago albeit a little slow but after i turned the rig off for a couple hours now i get this error. I erased the cpu.txt and it allowed me to be logged into the miner but then i realize only the cpu is mining. If i null the cpu i again get this error. any suggestions?

**Comment by psychocrypt** [comment]
please post the full output of the miner. remove amd.txt and cpu.txt and
start again

Am Do., 6. Dez. 2018, 04:47 hat thewalrusking <notifications@github.com>
geschrieben:

**Comment by thewalrusking** [comment]
'r' - results
'c' - connection
-------------------------------------------------------------------
Upcoming xmr-stak-gui is sponsored by:
   #####   ______               ____
 ##     ## | ___ \             /  _ \
#    _    #| |_/ /_   _   ___  | / \/ _   _  _ _  _ _  ___  _ __    ___  _   _
#   |_|   #|    /| | | | / _ \ | |   | | | || '_|| '_|/ _ \| '_ \  / __|| | | |
#         #| |\ \| |_| || (_) || \_/\| |_| || |  | | |  __/| | | || (__ | |_| |
 ##     ## \_| \_|\__, | \___/ \____/ \__,_||_|  |_|  \___||_| |_| \___| \__, |
   #####           __/ |                                                  __/ |
                  |___/   https://ryo-currency.com                       |___/

This currency is a way for us to implement the ideas that we were unable to in
Monero. See https://github.com/fireice-uk/cryptonote-speedup-demo for details.
-------------------------------------------------------------------
[2018-12-06 06:52:09] : Mining coin: monero
WARNING: AMD cannot load backend library: xmrstak_opencl_backend.dll
[2018-12-06 06:52:09] : WARNING: backend AMD (OpenCL) disabled.
[2018-12-06 06:52:09] : NVIDIA: try to load library 'xmrstak_cuda_backend_cuda10_0'
WARNING: NVIDIA Insufficient driver!
[2018-12-06 06:52:09] : NVIDIA: try to load library 'xmrstak_cuda_backend_cuda9_2'
WARNING: NVIDIA cannot load backend library: xmrstak_cuda_backend_cuda9_2.dll
WARNING: NVIDIA Insufficient driver!
[2018-12-06 06:52:09] : NVIDIA: try to load library 'xmrstak_cuda_backend'
WARNING: NVIDIA Insufficient driver!
[2018-12-06 06:52:09] : WARNING: backend NVIDIA disabled.
[2018-12-06 06:52:09] : WARNING: backend CPU disabled.
[2018-12-06 06:52:09] : ERROR: No miner backend enabled.
Press any key to exit.

**Comment by thewalrusking** [comment]
and i erase the amd.txt file and the cpu file and the cpu file repopulated but the amd file did not.

**Comment by Spudz76** [comment]
looks like `xmrstak_opencl_backend.dll` is missing, try to restore
it should exist wherever the configs are (working dir)

**Comment by psychocrypt** [comment]
yes download the miner again and add a exeption to your anti virus
software. The AV software removed the file.
